pkg/config/config.go

```diff
@@ -75,11 +77,40 @@ func (c *Config) ConfigureRules() {
 	}
 
 	logRuleset("default", rule.DefaultRules)
+	var excludedIndices []int
+
+RuleLoop:
+	for i, r := range c.Rules {
+		for _, ex := range c.ExcludeCategories {
+			// append and continue to next rule if category match found
+			if r.ContainsCategory(ex) {
+				excludedIndices = append(excludedIndices, i)
+				continue RuleLoop
+			}
+		}
 
-	for _, r := range c.Rules {
 		r.SetRegexp()
 		r.SetIncludeNote(c.IncludeNote)
 	}
+
+	// Remove excluded rules after done iterating through them
+	if len(c.ExcludeCategories) > 0 {
+		log.Debug().Strs("categories", c.ExcludeCategories).Msg("excluding categories")
+	}
+	for i, exIdx := range excludedIndices {
+		// every time a rule is removed, index of rules that come after it must be reduced by one
+		adjustedIdx := exIdx - i
+		log.Debug().Strs("categories", c.Rules[adjustedIdx].Options.Categories).Msg(fmt.Sprintf("rule \"%s\" excluded with categories", c.Rules[adjustedIdx].Name))
```

Comment:
nit: format - Is this possible to format? line 103 is very long. Possibly newline indent at `.Msg`?